tasks/caso-config.yaml

```diff
@@ -1,208 +1,220 @@
 ---
+- set_fact:
+    caso_conf: /usr/etc/caso/caso.conf
+    mapping_file: /usr/etc/caso/voms.json
+  when: ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat'
+
 - name: Create caso configuration directory
   file:
-    path: /etc/caso
+    path: "{{ item }}"
     state: directory
+  with:
+    - {{ caso_mapping_file | dirname }}
+    - {{ caso_conf_file | dirname }}
 
 - name: Configure voms.json
   template:
     src: voms.json.j2
-    dest: /etc/caso/voms.json
+    dest: {{ caso_mapping_file }}
+    owner: root
+    group: root
+    mode: 0644
+
 
-- name: Create /etc/caso/caso.conf
+- name: Create "{{ caso_conf_file }}"
   block:
```

Comment:
There seems to be a block of file creations when a variable is defined (oh my god, I sound like github_clippy already). I would use a loop here, and express the condition for the entire block.

Reply:
how to use a loop here? it's a bunch of variables being set in a single ini file

Reply:
so, assuming that the file is constant, the loop is over the "option" and "value". If you keep these in a var or a data structure, you can do something like : 

```yaml
- name: Configure the ini file
  path: "{{ caso_conf_file }}"
  section: "{{ item.section }}"
  option: "{{ item.option }}"
  value: "{{ item.value }}"
loop: "{{ ini_file_var }}"
```

Then your `ini_file_var` is like : 

```
---
- section: one
  option: best_option
  value: rad
- section: two
  option: compromise
  value: rad--
```

That way you can have one task, and easily separate actions from data - this will help maintain the codebase in the long run. What do you think?

Reply:
(adding for reference : https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#standard-loops)